src/tests/UnitTests/UI/TreeBaseTests.cs

```diff
@@ -606,6 +611,74 @@ public void ShouldPopulateTreeWithSingleEntryWithMetaInPath()
                 }
             });
         }
+
+        [Test]
+        public void ShouldCheckParentOfMetaFile()
```

Comment:
I know these are a pain to set up, but could we maybe do a test where we have
  - [X] Root
    - [x] Parent
      - [X] A
      - [X] B
      - [x] C

And then we uncheck `C` and see `Parent` and `Root` both go grey?